core/src/main/java/org/elasticsearch/discovery/zen/ping/unicast/UnicastZenPing.java

```diff
@@ -565,7 +565,8 @@ public void writeTo(StreamOutput out) throws IOException {
     }
 
     private PingResponse createPingResponse(DiscoveryNodes discoNodes) {
-        return new PingResponse(discoNodes.getLocalNode(), discoNodes.getMasterNode(), clusterName, contextProvider.nodeHasJoinedClusterOnce());
+        return new PingResponse(discoNodes.getLocalNode(), discoNodes.getMasterNode(), clusterName,
```

Comment:
+1 to what Nik said. The most important is that there is a single way so let's not stall this PR on the naming convention choice. Since people here seem to either not care or prefer the shorter version, then let's just rename getters to their shorter version and merge this PR?